src/lib/ppx_coda/versioned_type.ml

```diff
@@ -82,6 +90,98 @@ let validate_module_version module_version loc =
          begin with 0, got: \"%s\""
         module_version
 
+[%%if
```

Comment:
Doesn't need to be done now, but it might be better in the future to create a runtime flag to toggle the behaviour here and lift the optcomp into `run_ppx_coda`, so that we don't force a recompile of everything that depends on `ppx_coda` to run the check.

Reply:
The CI check only requires building `run_ppx_coda.exe`, which takes only a few seconds. The dune counter is 138 lines, as I recall.

Reply:
This is true for CI, but if we ever want to run it locally dune will end up throwing away the compiled versions of all files that depend on `ppx_coda`.